src/hooks/useSignup.js

```diff
@@ -0,0 +1,911 @@
+      return {
+        ...suggestedPlanData,
+        planList: [
+          ...suggestedPlanData?.plans?.original_plan?.plans || [],
+          ...suggestedPlanData?.plans?.suggested_plan?.plans || [],
+        ],
+      };
+    } catch (error) {
+      console.error(error);
+      return {};
+    }
+  };
+
+  /**
+ * @param {Array} subscriptions List of subscriptions of user
+ * @param {String} plan Base plan slug
+ * @returns {Promise<PlanExistenceObject>}
+ */
+  const validatePlanExistence = async (subscriptions, plan = '') => {
+    const planSlug = plan || BASE_PLAN;
+    try {
+      const planComparison = await getSuggestedPlan(planSlug, true);
+      const basePlan = planComparison?.original_plan;
+      const suggestedPlan = planComparison?.suggested_plan;
+
+      const hasBasePlan = subscriptions.some((s) => s?.plans?.[0]?.slug === basePlan?.slug);
+      const hasASuggestedPlan = subscriptions.some((s) => s?.plans?.[0]?.slug === suggestedPlan?.slug);
+
+      return {
+        basePlan,
+        suggestedPlan,
+        hasBasePlan,
+        hasASuggestedPlan,
+        allSubscriptions: subscriptions,
+      };
+    } catch (error) {
+      return {
+        basePlan: {},
+        suggestedPlan: {},
+        hasBasePlan: false,
+        hasASuggestedPlan: false,
+        allSubscriptions: subscriptions,
+      };
+    }
+  };
+
+  const handlePayment = async (data, disableRedirects = false) => {
+    const manyInstallmentsExists = selectedPlanCheckoutData?.how_many_months > 0 || selectedPlanCheckoutData?.period === 'FINANCING';
+    const isTtrial = ['FREE', 'TRIAL'].includes(selectedPlanCheckoutData?.type);
+
+    const getRequests = () => {
+      if (!isTtrial) {
+        return {
+          type: data?.type || checkoutData.type,
+          token: data?.token || checkoutData.token,
+          how_many_installments: data?.installments || selectedPlanCheckoutData?.how_many_months || undefined,
+          chosen_period: manyInstallmentsExists ? undefined : (selectedPlanCheckoutData?.period || 'HALF'),
+          coupons: checkoutData?.coupons,
+        };
+      }
+      return {
+        type: data?.type || checkoutData.type,
+        token: data?.token || checkoutData.token,
+      };
+    };
+
+    try {
+      const requests = getRequests();
+      const response = await bc.payment().pay({
+        ...requests,
+        conversion_info: {
+          ...userSession,
+        },
+      });
+
+      const transactionData = await response.json();
+
+      if (transactionData?.status === 'FULFILLED') {
+        setSubscriptionProcess({
+          status: subscriptionStatusDictionary.PREPARING_FOR_COHORT,
+          id: dateProps?.id,
+          slug: dateProps?.slug,
+          plan_slug: dateProps?.plan?.slug,
+          academy_info: dateProps?.academy,
+        });
+
+        let currency = 'USD';
+        let simplePlans = [];
+        if (cohortPlans) {
+          currency = cohortPlans[0]?.plan?.currency?.code || 'USD';
```

Comment:
Use if statement to validate the value we want to assign otherwise don't modify it